roles/openshift_facts/library/openshift_facts.py

```diff
@@ -430,6 +408,46 @@ def set_aggregate_facts(facts):
 
     return facts
 
+def set_deployment_facts_if_unset(facts):
```

Comment:
@detiber We'd discussed moving all the deployment_type related facts into one function to localize things. However pylint limits branching to 12 branches. If we add more in the future we'll either have to add pylint annotations to override that for this function or we'll have to break it up into smaller functions.